### DISCUSSION TITLE
Multiple faces not being detected

### DISCUSSION THREAD

**Discussion by BurningTheSky** [photoprism/photoprism#2042] [Q&A] [answered]
I've found a number of pictures where faces which are close together don't get detected, only 1 does. It also seems to happen when there are 3+ faces close together, only 1 is detected. Is there any way to relax the configuration to make these multiple faces be detected even though it might risk false positives?

**Comment by graciousgrey** [comment, marked as answer]
Facial detection and facial recognition is never perfect. Even Google does not detect all faces. It took us lot's of time to optimize the parameters. 

All config options for facial recognition are documented here https://docs.photoprism.app/getting-started/config-options/#facial-recognition

**Comment by BurningTheSky** [comment]
I understand that, but I am seeing a fairly high hit rate of missed faces when in proximity to other faces, ones that aren't missed in say DigiKam or the Nextcloud plugin. I generally prefer how photoprism does it but these missed faces and children not being tagged does cause some issues in using it.

**Reply by graciousgrey** [reply]
A lot of improvements are planned regarding facial detection/recognition: https://github.com/photoprism/photoprism/issues?q=is%3Aissue+is%3Aopen+people+in%3Atitle

The detection of faces will be improved as well at some point :)

We just can't work on everything at the same time given our current funding (https://github.com/sponsors/photoprism)

**Reply by BurningTheSky** [reply]
I completely understand! I think the work done so far is amazing, especially as it is free to use!

**Reply by BurningTheSky** [reply]
I think it might be a coming soon feature but is it possible to manually add missing people at the moment?

**Reply by graciousgrey** [reply]
That's also on our to do: https://github.com/photoprism/photoprism/issues/1548 :)
